packages/core/src/utils/checkpoint.ts

```diff
@@ -102,19 +142,12 @@ export const isCheckpointGreaterThan = (a: Checkpoint, b: Checkpoint) => {
     return a.blockTimestamp > b.blockTimestamp;
   if (a.chainId !== b.chainId) return a.chainId > b.chainId;
   if (a.blockNumber !== b.blockNumber) return a.blockNumber > b.blockNumber;
-
-  // If both logIndex are defined, compare normally.
-  if (a.logIndex !== undefined && b.logIndex !== undefined) {
-    return a.logIndex > b.logIndex;
-  }
-  // If both are undefined, the checkpoints are equal, so a is not greater than b.
-  if (a.logIndex === undefined && b.logIndex === undefined) {
-    return false;
-  }
-  // If only a is undefined, it's considered greater.
-  if (a.logIndex === undefined) return true;
-  // Otherwise b is undefined and a is defined, so b is greater.
-  else return false;
+  if (a.transactionIndex !== b.transactionIndex)
```

Comment:
I wonder if it's easier to encode both checkpoints and then compare the strings lexicograpically, OR encode, convert to bigint, and compare.